### PR TITLE
feat(api-qraphql-flow): DatoCMS setup

### PR DESCRIPTION
**Pull request by emberglaz**
* add @apollo/client
* add Firebase email verify
* add initLocalizationSaga flow
* add initQuestSaga flow
* add saveTranslations Redux flow
* add saveQuests Redux flow
* add graphql client for  DatoCMS

### CODE REVIEW

File: app/redux/slices/settingsSlice.ts [hunk 1]
```diff
@@ -1,14 +1,14 @@
 import { createSlice } from '@reduxjs/toolkit';
 
 import { ISetSettingsDataPayload, ISettingsData } from '../types';
-
+import CONFIG from '../../config/env';
 export interface ISettingsState {
```
Comment (Reykjavik151) [inline app/redux/slices/settingsSlice.ts]: Space line between` import` statement and actual code.

File: app/sagas/localization.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    const translations: Record<string, string> = yield call(
+      getTranslations,
+      language,
+    );
+    if (translations) {
+      yield put(
+        cacheSlice.actions.saveTranslations({ [language]: { translations } }),
+      );
+    } else {
+      const loadTranslations: Record<string, string> = yield select(
```
Comment (Reykjavik151) [inline app/sagas/localization.ts]: rename to `reduxTranslations`

File: app/screens/Onboarding/Parents/SignIn/SelectUser/SelectUser.tsx [hunk 3]
```diff
@@ -27,7 +26,9 @@ export const SelectUserScreen: React.FC<ISelectUserScreenProps> = ({
       <BottomButtonView
         buttonTitle={t('buttons.manage_profiles')}
         // TODO: change noop to real function
-        onSubmit={_.noop}>
+        onSubmit={() => {
+          trackButtonPress('manage');
+        }}>
```
Comment (Reykjavik151) [inline app/screens/Onboarding/Parents/SignIn/SelectUser/SelectUser.tsx]: Please, do not create arrow functions in JSX thread.
Create function before and wrap it with `useCallback`

File: app/screens/Onboarding/Parents/WelcomeParent/WelcomeParent.tsx [hunk 4]
```diff
@@ -22,6 +23,7 @@ export const WelcomeParentScreen: React.FC<IWelcomeParentScreenProps> = ({
 
   const onLoginPress = useCallback(() => {
     navigation.navigate('Login');
+    trackButtonPress('Login');
   }, [navigation]);
```
Comment (Reykjavik151) [inline app/screens/Onboarding/Parents/WelcomeParent/WelcomeParent.tsx:27]: We should patch our ExtendedButton with analytic for ALL buttons, do not do it manually.

File: app/services/graphql/graph.types.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+export const getAllTranslationsQuery = (locale: string) => gql`
+  query getAllTranslations {
+    allTranslations(locale: ${locale}) {
+      id
+      label
+      text
+    }
+  }
+`;
```
Comment (Reykjavik151) [inline app/services/graphql/graph.types.ts]: It isn't `types` as `filename` says. Rename the file.

File: app/services/graphql/index.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+  constructor(baseURL = CONFIG.API_URL, token = CONFIG.API_KEY) {
+    this.client = new ApolloClient({
+      uri: baseURL,
+      cache: new InMemoryCache(),
+      headers: {
+        'X-Api-Version': '3',
+        Accept: 'application/json',
+        authorization: `Bearer ${token}`,
+      },
```
Comment (Reykjavik151) [inline app/services/graphql/index.ts]: Make constants for header and spread some of them here.

File: app/services/localization/index.ts [hunk 7]
```diff
@@ -1,6 +1,7 @@
-import i18n from 'i18next';
 import { initReactI18next } from 'react-i18next';
+import i18n from 'i18next';
 
+import Api from '../graphql/index';
```
Comment (Reykjavik151) [inline app/services/localization/index.ts]: redundant `/index` suffix

File: ios/SaferPlace.xcodeproj/xcshareddata/xcschemes/SaferPlace.xcscheme [hunk 8]
```diff
@@ -60,6 +60,12 @@
             ReferencedContainer = "container:SaferPlace.xcodeproj">
          </BuildableReference>
       </BuildableProductRunnable>
+      <CommandLineArguments>
+         <CommandLineArgument
+            argument = "-FIRDebugEnabled"
+            isEnabled = "YES">
+         </CommandLineArgument>
+      </CommandLineArguments>
```
Comment (Reykjavik151) [inline ios/SaferPlace.xcodeproj/xcshareddata/xcschemes/SaferPlace.xcscheme]: Should we keep it for production version?

**Review by Reykjavik151 — CHANGES_REQUESTED** [review]

File: app/screens/Onboarding/Parents/WelcomeParent/WelcomeParent.tsx (hunk 4 above)
Comment (emberglaz) [reply app/screens/Onboarding/Parents/WelcomeParent/WelcomeParent.tsx:27]: in next pr

File: app/services/datocms/index.ts [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+      if (response.data.allTranslations.length) {
+        result = result.concat(response.data.allTranslations);
+      } else {
+        const translations = translationsToDictionary(result);
+        return translations;
+      }
+
+      offset += PAGINATION_STEP;
+    }
+  }
```
Comment (Reykjavik151) [inline app/services/datocms/index.ts]: change to arraow function

File: app/services/datocms/index.ts [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+
+      if (response.data.allQuestLines.length) {
+        result = result.concat(response.data.allQuestLines);
+      } else {
+        return result;
+      }
+
+      offset += PAGINATION_STEP;
+    }
+  }
```
Comment (Reykjavik151) [inline app/services/datocms/index.ts]: change to arrow function

File: app/services/datocms/index.ts [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+        return result;
+      }
+
+      offset += PAGINATION_STEP;
+    }
+  };
+}
+export const apiInstance = new DatoCMSApi();
+
+export default DatoCMSApi;
```
Comment (Reykjavik151) [inline app/services/datocms/index.ts]: do not export anything, only apiInstance

File: app/services/datocms/index.ts [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+        result = result.concat(response.data.allQuestScreens);
+      } else {
+        return result;
+      }
+
+      offset += PAGINATION_STEP;
+    }
+  };
+}
+export const apiInstance = new DatoCMSApi();
```
Comment (Reykjavik151) [inline app/services/datocms/index.ts]: 1. add spaceline
2. rename to datocmsApiInstance

**Review by Reykjavik151 — CHANGES_REQUESTED** [review]

**Pull request merged**